lib/suse/connect/api.rb

```diff
@@ -220,7 +220,16 @@ def system_migrations(auth, products, target_base_product: nil, kind:)
           online: '/connect/systems/products/migrations',
           offline: '/connect/systems/products/offline_migrations'
         }
-        @connection.post(endpoints.fetch(kind), auth: auth, params: payload)
+        result = @connection.post(endpoints.fetch(kind), auth: auth, params: payload)
+        # sort migration targets by product and version
+        if result.body.is_a?(Array)
+          result.body.each do |array|
+            array.sort_by! do |product|
+              [ products.map(&:identifier).index(product['identifier']) || 9000, -product['version'].to_f ]
```

Comment:
In general I dislike magic constants like `9000`. I know the `index` will never hit `9000` but it's hard to understand when looking at the source after some time. Maybe define a constant like `PRODUCT_NOT_INSTALLED` and add the `9000` there or find a better solution to handle the `nil` case.

Reply:
In theory the `version` should always be a string which can be converted to a float. Until this is not the case. Maybe add at least a error handling in case it will fail and throw an exception. Maybe this is just `nitpicking` but you know you never know what will happen!